pkg/actuators/machine/actuator.go

```diff
@@ -641,7 +638,7 @@ func (a *Actuator) updateStatus(machine *machinev1.Machine, instance *ec2.Instan
 	// we get a public IP populated more quickly.
 	if awsStatus.InstanceState != nil && *awsStatus.InstanceState == ec2.InstanceStateNamePending {
```

Comment:
we should probably move this check up outside of this function, so the func does not error, so the patch request takes place so the machine and the status are updated nevertheless